tests/ui/issues/issue-5883.stderr

```diff
@@ -15,22 +15,6 @@ help: function arguments must have a statically known size, borrowed types alway
 LL |     r: &dyn A + 'static
    |        +
 
-error[E0277]: the size for values of type `(dyn A + 'static)` cannot be known at compilation time
```

Comment:
This error goes away because we are now able to successfully deduplicate the duplicated sized obligation that results here.

I'm 99% sure that this is an implementation detail that has to do with the way that the `ObligationForest` stores and deduplicates the obligations that it knows it has processed already, and the way that the `GatherLocalsVisitor` registers `Sized` obligations for parameters. 😅

Reply:
remove duplicate sized errors with this one simple trick (removing the hack to remove duplicate sized errors)